libtransmission/peer-mgr.cc

```diff
@@ -466,6 +466,10 @@ class tr_swarm
             peer_info.found_at(from);
             peer_info.set_pex_flags(flags);
         }
+        else
```

Comment:
```suggestion
        else if (is_connectable)
```

We shouldn't count incoming peers with unknown listening port as "known peers".

Reply:
You also need an extra line here to decrement the known peers stat before we housekeep inactive peers.

https://github.com/transmission/transmission/blob/9ef0d71aa965fae6a7bf9ee595ba2b90d2ec9da0/libtransmission/peer-mgr.cc#L348-L351

Something like:

```cpp
if (peer_info.is_inactive(now)) 
{
    --this->stats.known_peer_from_count[iter->second.from_first()];
    iter = connectable_pool.erase(iter); 
}
```

Reply:
My my ... there are 3 `connectable_pool.extract`, 2 `connectable_pool.erase` and 1 `connectable_pool.insert` ... I need to figure this out in a good way.

Reply:
Good catch...! I forgot about those.

Sorry about those pile of mess, I couldn't think of a better way when I wrote that code... I'll help you figure out where else you need to put increment/decrement statements.

Reply:
I may have figured it out.

Reply:
For `on_got_port*()`, you only need to increment the stat inside this block.

https://github.com/transmission/transmission/blob/7d1a32c4719f0b54ede1a2046c2342608c9bbb35/libtransmission/peer-mgr.cc#L821-L824

 All other code paths should not affect the total number of known peers.

Reply:
I think you've got the right idea.

See my follow-up comments https://github.com/transmission/transmission/pull/4900#pullrequestreview-1710011333, I think these will be the final touches before this PR is good to go...!